lib/Doctrine/ORM/Persisters/Collection/OneToManyPersister.php

```diff
@@ -182,10 +182,15 @@ public function removeElement(PersistentCollection $collection, $element)
             return false;
         }
 
+        $this->uow->scheduleOrphanRemoval($element);
+
+        return true;
+        /*
         return $this
```

Comment:
In the context of an `EXTRA_LAZY` association, an immediate delete seems more correct here, if I read it correctly.